pilot/pkg/networking/core/v1alpha3/accesslog.go

```diff
@@ -185,13 +187,18 @@ func buildAccessLogFromTelemetry(push *model.PushContext, spec *model.LoggingCon
 		var al *accesslog.AccessLog
 		switch prov := p.Provider.(type) {
 		case *meshconfig.MeshConfig_ExtensionProvider_EnvoyFileAccessLog:
-			al = buildEnvoyFileAccessLogHelper(prov.EnvoyFileAccessLog)
+			// For built-in provider, fallback to Mesh Config for formatting options.
+			if p.Name == defaultEnvoyAccessLogProvider {
```

Comment:
should only fallback to `MeshConfig` when `p.Name == "envoy" and prov.EnvoyFileAccessLog.LogFormat==nil` ?
cc @ramaraochavali @douglas-reid

Reply:
Is it the case of user creating access log provider with the same name?